### PR TITLE
fix(47076):Fix error term of declaration in modules

### LINKED ISSUE #47076: Improve non-top-level import/export error, especially for JS
ES imports and exports can only be used at the top level of a module. This is illegal:

```ts
function container() {
  import 'fs'
  export { container }
  namespace N { }
}
```

The current errors for these three statements are vague and, for JS, contain irrelevant terms:

Actual:

(1) "An import declaration can only be used in a namespace or module."
(2) "An export declaration can only be used in a module."
(3) "A namespace declaration is only allowed in a namespace or module."

Expected:

(1) When the node is in a JS file, "An import declaration can only be used at the top level of a module."
  Otherwise, "An import declaration can only be used at the top level of a namespace or module."
(2) When the node is in a JS file, "An export declaration can only be used at the top level of a module."
  Otherwise, "An export declaration can only be used at the top level of a namespace or module."
(3) "A namespace declaration is only allowed at the top level of a module."

Implementation:
`checkGrammarModuleElementContext` issues these errors. I don't know whether it's better to make it smarter or just avoid calling it for case (1).

### PR DESCRIPTION
**Pull request by islandryu**
Fixes #47076

I fix error term of declaration in modules.
- "An import declaration can only be used in a namespace or module." →
  - When the node is in JS file, "An import declaration can only be used at the top level of a module."
  - Otherwise, "An import declaration can only be used at the top level of a namespace or module."

- "An export declaration can only be used in a module." →
  - "An export declaration can only be used at the top level of a module."

- "A namespace declaration is only allowed in a namespace or module." →
  - "A namespace declaration is only allowed at the top level of a module."

### CODE REVIEW

**Comment by DanielRosenwasser** [comment]
Why isn't it

> A namespace declaration is only allowed at the top level of **a namespace or** module.

?

**Comment by islandryu** [comment]
@DanielRosenwasser 
Indeed, it should be that way.
Fixed.

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -39795,7 +39795,7 @@ namespace ts {
         }
 
         function checkImportDeclaration(node: ImportDeclaration) {
-            if (checkGrammarModuleElementContext(node, Diagnostics.An_import_declaration_can_only_be_used_in_a_namespace_or_module)) {
+            if (checkGrammarModuleElementContext(node,isInJSFile(node) ? Diagnostics.An_import_declaration_can_only_be_used_at_the_top_level_of_a_module : Diagnostics.An_import_declaration_can_only_be_used_at_the_top_level_of_a_namespace_or_module)) {
```
Comment (DanielRosenwasser) [inline src/compiler/checker.ts:39798]:
```suggestion
            if (checkGrammarModuleElementContext(node, isInJSFile(node) ? Diagnostics.An_import_declaration_can_only_be_used_at_the_top_level_of_a_module : Diagnostics.An_import_declaration_can_only_be_used_at_the_top_level_of_a_namespace_or_module)) {
```

Comment (islandryu) [reply src/compiler/checker.ts:39798]: When I hit commit suggestion, for some reason I got the message "This diff has recently been updated.So I updated it locally.

**Review by sandersn — APPROVED** [review]

**Pull request merged**